File: 001 - TwoSum/Solution2.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}
        for i, value in enumerate(nums):
            complement = target - value
            if complement in seen:
                return (seen[complement], i)
            seen[value] = i
        return None
                    

File: 001 - TwoSum/Solution2.cpp
class Solution {
public:
    std::vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> numMap;
        for (int i = 0; i < nums.size(); i++){
            int diff = target - nums[i];
            auto iterator = numMap.find(diff);
            if (iterator != numMap.end()){
                return {iterator->second, i};
            }
            numMap[nums[i]] = i;
        }

        return {};
    }
};
